### ISSUE TITLE
Feature Request : uncheck items already checked instead of reating new ones

### ISSUE THREAD

**Issue by toxic0berliner** [baggachipz/tinylist#49]
in short : when trying to add an alement to a list, if there are some items hidden in the "checked items" section that could match the current entry, offer a way to simply uncheck them instead of the current entry.
Just like google keep is doing ;)

For context: 
Using this or keep to track my groceries list, I tend to reuse the same list week after week and I also tend to buy similar things over and over again each week.
So for example, in google keep, when I type "toma" it offers me to bring back the item I cheked last week that is "6 Tomatoes 🍅", I just tap on the checked checkmark it offers me and it removes my "toma" entry replacing it with the nice formatted one with all the details I like :D 

This also works in tandem with another feature I realy like wich is sublevels in the checklist, for example : 
 - [ ] Veggies 🌿
   - [ ] Tomatoes 🍅
   - [ ] Salad 🥬
 - [ ] Eggs 🥚

Cheking veggies  when doing the groceries would also check both Tomatoes and Salad but not eggs.
And when trying to add "vegg" it would offer me to uncheck "Veggies 🌿" which would also bring back Tomatoes and Salad as unchecked in the list.

Quite the time saver when preparing and doing the groceries as one can add and remember details in the list so that if the other is doing the groceries details are in there and not forgotten :D

**Comment by baggachipz** [comment]
So this actually looks like two feature requests to me:

1. Checklist sub-items
2. Smart autocomplete/undelete

The first is much more doable and I should be able to get that done soon. I've received that request before so it's probably time to do it. The second is much more intensive because I currently rely on the browser/system to do autocomplete/autocorrect and it works "well enough" for most use cases and keeps complexity low. I would have to override that behavior and while possible, isn't trivial. But I have to admit, it's a good feature request and would be quite useful. 

Before I add more features, I need to get tinylist upgraded to use Quasar 2/Vue 3 (the framework which underpins things) as the previous version I use now is getting out of date. Once I do that, I should be able to work on new features. Thanks for your feedback and suggestions!

**Comment by toxic0berliner** [comment]
Yeah, I'm sneaky like that slipping in a second feature request on top of the first one :D 
Disguised as it was maybe one would have done hours and hours of coding to tweak autocomplete without realizing sub-items are a much easyer target, only because I didn't start with it...

Sorry, joke aside, you're fully correct in your understanding, and I must say the app looke real good and I'm amazed at how fast and thouroughly you answer here !

Thanks already for all the hard work done and in advance for what's coming ! Good luck with Quasar/Vue upgrade !
